pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/ColumnMetadata.java

```diff
@@ -90,7 +92,40 @@ default boolean isMinMaxValueInvalid() {
   @Nullable
   Set<Integer> getPartitions();
 
-  Map<IndexType<?, ?, ?>, Long> getIndexSizeMap();
+  /**
+   * If exists, returns size of index for given index type; otherwise returns -1.
+   * @param type index type
+   */
+  default long getIndexSizeFor(IndexType type) {
```

Comment:
If this is just for testing purpose, consider annotating it with `@VisibleForTesting` and add javadoc warning future developers to not use it because of the overhead

Reply:
Right now it's for testing (because production usage for the collection is basically iteration) but, in general, it's better api. I've added the annotation anyway. As for the overhead - I don't think there's anything to worry about. The list is a few items-big at most, so lookup should is likely faster in the older map implementation.